c++/triqs/stat/accumulator.hpp

```diff
@@ -344,9 +357,44 @@ namespace triqs::stat {
     ///
     /// @param lin_bin_capacity The number of measurements the linear part will average together in a single bin, before starting a new bin.
     ///
-    accumulator(T const &data_instance, int n_log_bins_max = 0, int n_lin_bins_max = 0, int lin_bin_capacity = 1)
+    accumulator(T const &data_instance, int n_log_bins_max = 0, int n_lin_bins_max = 0, int lin_bin_capacity = 1,
+                std::function<void(std::vector<T> const & /* lin_bins */)> callback = {})
        : log_bins{data_instance, n_log_bins_max}, //
-         lin_bins{data_instance, n_lin_bins_max, lin_bin_capacity} {}
+         lin_bins{data_instance, n_lin_bins_max, lin_bin_capacity, callback} {
+
+      // enable default if callback has not been defined and log binning is active (such that Qk[0] and Mk[0] are available)
+      if ((!callback) && (n_log_bins_max > 0)) {
+        callback = [&](std::vector<T> const &bins) {
+          using var_t = get_real_t<T>;
+
+          // variance of the whole time series
+          T mean_no_binning    = log_bins.Mk[0];
+          var_t var_no_binning = log_bins.Qk[0] / log_bins.count - make_real(nda::hadamard(nda::conj(mean_no_binning), mean_no_binning));
+
+          // mean of the binned data
+          T mean_with_binning = bins[0];
+          for (auto i : range(1, bins.size())) mean_with_binning += bins[i];
+          mean_with_binning /= bins.size();
+
+          // variance of the binned data
+          T delta                = bins[0] - mean_with_binning;
+          var_t var_with_binning = make_real(nda::hadamard(nda::conj(delta), delta));
+
+          for (auto i : range(1, bins.size())) {
+            delta = bins[i] - mean_with_binning;
+            var_with_binning += make_real(nda::hadamard(nda::conj(delta), delta));
+          }
+
+          var_with_binning /= bins.size();
```

Comment:
Please use the `mean_and_err` function from the `mean_and_err.hpp` header here.
Adjust it accordingly to use `hadamard`.

Reply:
Reopened as #967. I would propose to not use the `mean_and_err` function since we want the variance of the sample not the error on the mean in this case (we thus do not need to assume that `sqrt(T)` is defined).